specs/Picker-API_spec.md

```diff
@@ -0,0 +1,121 @@
+# Background
+
+The current FilePicker API for UWP has several limitations. Some of the top limitations 
+are as described below:
+
+**1. Unable to pick files and folders in a single instance of the Picker dialog**
+
+   The Picker dialog allows developers to specify picker mode to be single file or multiple
+   files. It does not however, allow developers to specify a mode to pick both files and folders.
+   There has been feedback indicating that this behaviour is limiting, requiring developers
+   to build their own picker experience. 
+
+**2. Unable to pick multiple folders.**
+
+   The Picker dialog (FolderPicker) does not allow users to pick multiple folders. The developer
+   experience currently is to select a single folder and repeat the dialog for more folders. This 
+   is a less than ideal experience for application developers.
+
+**3. Unable to specify an arbitrary start location for the picker.**
+
+   The existing File/Folder pickers do not allow the developer to specify an arbitrary start location (even if the app has access). Currently, the start location is limited to a set of predefined enum values.
+
+The Reunion picker dialog addresses the above issues. This Reunion API will be based upon the below
+APIs in the Windows SDK. The Reunion Picker API will supercede the existing APIs and add additional methods
+for the functionality: File and folder picker, Multiple folder picker.
+
+- Windows.Storage.Pickers.FileOpenPicker (https://docs.microsoft.com/en-us/uwp/api/Windows.Storage.Pickers.FileOpenPicker)
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The Project Reunion APIs will supercede the Windows APIs.
+
+# Examples
+## Show file and folder picker
+```c#
+    var picker = new StorageItemPicker();
+    picker.FiltersAndDescriptions.Insert("Images", "*.jpg, *.png");
+    picker.StartLocation = startLocation;
+    picker.PickerKinds = StorageItemPickerKinds.Folder | StorageItemPickerKinds.FileOpen;
+    // multi-select will be automatically set
```

Comment:
Multi-select is the default? Is that what applications expect? It defaults to `false` in the .NET pickers.

Reply:
MultiSelect is being automatically set since the PickerKind is both files and folders. Multiselect is false by default.

---
In reply to: [500443235](https://github.com/microsoft/ProjectReunion/pull/99#discussion_r500443235) [](ancestors = 500443235)

Reply:
That doesn't logically follow though... it's legitimate to want to pick a single file or a single folder. Also, having one property change value based on another property's value is against guidelines. For example, the following two bits of code **should** be equivalent:

```
picker.PickerKinds = Folder | FileOpen;
picker.Multiselect = false;
```

and

```
picker.Multiselect = false;
picker.PickerKinds = Folder | FileOpen;
```

But they are not equivalent; the first one works as expected, but the second one does not.

Reply:
Updated to false by default. No automatic setting of the property.

---
In reply to: [500446491](https://github.com/microsoft/ProjectReunion/pull/99#discussion_r500446491) [](ancestors = 500446491,500443235)